src/is_subset.jl

```diff
@@ -887,3 +887,75 @@ function ⊆(X::LazySet{N}, C::Complement{N}, witness::Bool=false
           )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
     return isdisjoint(X, C.X, witness)
 end
+
+
+# --- CartesianProduct ---
+
+
+"""
+    ⊆(X::CartesianProductArray{N}, Y::CartesianProductArray{N},
+      witness::Bool=false; check_block_equality::Bool=true
+     )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
+
+Check whether a Cartesian product of finitely many convex sets is contained in
+the another Cartesian product of finitely many convex sets, and otherwise
+optionally compute a witness.
+
+### Input
+
+- `X`       -- Cartesian product of finitely many convex sets
+- `Y`       -- Cartesian product of finitely many convex sets
+- `witness` -- (optional, default: `false`) compute a witness if activated
+- `check_block_equality` -- (optional, default: `true`) flag for checking that
+               the block structure of the two sets is identical
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``X ⊆ Y``
+* If `witness` option is activated:
+  * `(true, [])` iff ``X ⊆ Y``
+  * `(false, v)` iff ``X \\not\\subseteq Y`` and
+    ``v ∈ X \\setminus Y``
+
+
+### Notes
+
+This algorithm requires that the two Cartesian products share the same block
+structure.
+Depending on the value of `check_block_equality`, we check this property.
+
+### Algorithm
+
+We check for inclusion for each block of the Cartesian products.
+"""
+function ⊆(X::CartesianProductArray{N},
+           Y::CartesianProductArray{N},
+           witness::Bool=false;
+           check_block_equality::Bool=true
+          )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
+    aX = array(X)
+    aY = array(Y)
+    if check_block_equality && !same_block_structure(aX, aY)
+        throw(ArgumentError("this inclusion check requires Cartesian products" *
+                            "with the same block structure"))
+    end
+
+    for i in 1:length(aX)
+        result = ⊆(aX[i], aY[i], witness)
+        if !witness && !result
+            return false
+        elseif witness && !result[1]
```

Comment:
(side note: this reads a bit odd, that in one `if` one uses `!result` and in the second indexes the reuslt, `!result[1]`, but it's a consequence of the return type of inclusion checks which is either a `Bool` or a tuple `(::Bool, ::Vector)`)

Reply:
Yes, but what can I do?

Reply:
i think that it is good like that!

> Yes, but what can I do?

(writing an outer case `if witness . . .` and repeating the line `result = ⊆(aX[i], aY[i], witness)`, of course (not a big deal) is one way, otherwise with dispatch..).